dragonchain/lib/dto/bnb.py

```diff
@@ -275,36 +275,47 @@ def _fetch_account(self):
             _log.warning("[BINANCE] May have been a 500 Bad Request or a 404 Not Found.")
             raise exceptions.InterchainConnectionError("[BINANCE] Error fetching metadata from 'account' endpoint.")
 
-    def _build_transaction_msg(self, transaction_payload: str) -> Dict:
-        # cannot send an amount of 0; send funds to yourself and avoid hardcoding a dummy recipient address
-        inputs = {"address": self.address, "coins": [{"amount": 1, "denom": "BNB"}]}
-        outputs = {"address": self.address, "coins": [{"amount": 1, "denom": "BNB"}]}
-        response = self._fetch_account()
+    def _build_transaction_msg(self, raw_transaction: Dict[str, Any]) -> Dict:
+        """Build a formatted transaction for this network from the base parameters
+        Args:
+            amount (int): amount of token in transaction
+            to_address (str): hex of the address to send to
+            symbol (str): the exchange symbol for the token
+            memo (str): string of data to publish in the transaction
+        Returns:
+            dict of the constructed transaction
+        """
+        amount = raw_transaction["amount"]
+        if amount <= 0:
+            raise exceptions.BadRequest("[BINANCE] Amount in transaction cannot be less than or equal to 0.")
 
+        symbol = raw_transaction["symbol"]
```

Comment:
you assume symbol exists, but it's not required in your schema

Reply:
It's not required, but it is defaulted to "BNB" if it is not provided -- see `dragonchain_client.py`.

Reply:
yeah, but the sdk isn't how you should rely on a default. What if someone else writes a dragonchain client? Now the Dragonchain can return a 500 error if you don't know that you have to provide a symbol, which isn't required by the schema on Dragonchain.

You would expect not providing some part of the schema to result in a 400, not an unhandled 500.

Defaulting a value in the SDK is a nice-to-have, but is not a substitute for making sure the back-end accepts and handles good/bad/missing values.

An api is a user interface too. You definitely shouldn't make the assumption that everyone will interact with the Dragonchain via only our sanctioned SDKs, and the backend should be able to handle bad input.

So we need to either require symbol in the schema, or default it on the back-end if it's not provided.

Reply:
Made sure that `_build_transaction_msg()` function in the backend provided the proper defaults.  Removed defaults from SDK, feels redundant there.